src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.ManagedNtlm.cs

```diff
@@ -218,24 +218,32 @@ private unsafe struct NtChallengeResponse
             public override IIdentity RemoteIdentity => throw new InvalidOperationException();
             public override System.Security.Principal.TokenImpersonationLevel ImpersonationLevel => System.Security.Principal.TokenImpersonationLevel.Impersonation;
 
-            public ManagedNtlmNegotiateAuthenticationPal(NegotiateAuthenticationClientOptions clientOptions)
+            private ManagedNtlmNegotiateAuthenticationPal(NegotiateAuthenticationClientOptions clientOptions)
             {
-                Debug.Assert(clientOptions.Package == NegotiationInfoClass.NTLM);
-
                 _credential = clientOptions.Credential;
-                if (string.IsNullOrWhiteSpace(_credential.UserName) || string.IsNullOrWhiteSpace(_credential.Password))
-                {
-                    // NTLM authentication is not possible with default credentials which are no-op
-                    throw new PlatformNotSupportedException(SR.net_ntlm_not_possible_default_cred);
-                }
-
                 _spn = clientOptions.TargetName;
                 _channelBinding = clientOptions.Binding;
                 _protectionLevel = clientOptions.RequiredProtectionLevel;
 
                 if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, $"package={clientOptions.Package}, spn={_spn}, requiredProtectionLevel={_protectionLevel}");
             }
 
+            public static new NegotiateAuthenticationPal Create(NegotiateAuthenticationClientOptions clientOptions)
+            {
+                Debug.Assert(clientOptions.Package == NegotiationInfoClass.NTLM);
+
+                if (clientOptions.Credential == CredentialCache.DefaultNetworkCredentials ||
+                    string.IsNullOrWhiteSpace(clientOptions.Credential.UserName) ||
```

Comment:
I think this should be an `&&` between username and password.

Reply:
It was always `||` in all versions on .NET. It's not like empty user name with non-empty password, or vice versa, makes any sense.

Reply:
it's not? I've seen services that only require password with empty username.

Reply:
I don't think that's valid in NTLM or Kerberos. It's valid in Basic authentication.